src/test/java/com/microsoft/azure/vmagent/test/ITAzureVMManagementServiceDelegate.java

```diff
@@ -198,6 +198,55 @@ public void createDeploymentWithPrivateIPTest() throws Exception {
 
     }
 
+    @Test
+    public void createDeploymentWithPurchasePlan() throws Exception {
+        testEnv.azureImagePublisher = "kali-linux";
+        testEnv.azureImageOffer = "kali-linux";
+        testEnv.azureImageSku = "kali";
+
+        AzureVMAgentCleanUpTask.DeploymentRegistrar deploymentRegistrar = mock(AzureVMAgentCleanUpTask.DeploymentRegistrar.class);
+        when(deploymentRegistrar.getDeploymentTag()).thenReturn(new AzureUtil.DeploymentTag("some_tag/123"));
+        AzureVMDeploymentInfo deploymentInfo;
+        deploymentInfo = createDefaultDeployment(1, deploymentRegistrar);
+
+        verify(deploymentRegistrar).registerDeployment(null, testEnv.azureResourceGroup, deploymentInfo.getDeploymentName(), null);
+        Network actualVNet = null;
+        StorageAccount actualStorageAcc = null;
+        try {
+            actualVNet = azureClient.networks().getByResourceGroup(testEnv.azureResourceGroup, "jenkinsarm-vnet");
```

Comment:
any place to put all constants to?

Reply:
not yet, all the test cases are like this, it is better to reconstruct all of them next time.

Reply:
agree. PR approved already.